### ISSUE TITLE
inconsistent class score computation in FCOS codes

### ISSUE THREAD

**Issue by HawkRong** [open-mmlab/mmdetection#3382]
I noticed the class score computation in `FCOSHead`'s `_get_bbox_single` is different from the one in `forward_single`.
In `_get_bbox_single`:
`
scores = cls_score.permute(1, 2, 0).reshape(
                -1, self.cls_out_channels).sigmoid()
`
where `sigmoid` is called;
while in `forward_single`:
`
cls_score, bbox_pred, cls_feat, reg_feat = super().forward_single(x)
`
where the `super().forward_single` is defined in `AnchorFreeHead` as:
`
cls_score = self.conv_cls(cls_feat)
`
where no `sigmoid` is called.
This difference in class score computation is confusing. Is there any explanation on this inconsistency? Or is this an error?

**Comment by hellock** [comment]
The output of `forward_single` is used for computing losses, which requires logits. The output of `_get_bbox_single` is bbox scores, which is within [0, 1].

**Comment by HawkRong** [comment]
> The output of `forward_single` is used for computing losses, which requires logits. The output of `_get_bbox_single` is bbox scores, which is within [0, 1].

Do you mean that the sigmoid computation is included in the `FocalLoss` module? I noticed the class score loss computation is traced to:
`from mmcv.ops import sigmoid_focal_loss as _sigmoid_focal_loss`
so, is the sigmoid computation done in the `mmcv.ops.sigmoid_focal_loss`?

**Comment by hellock** [comment]
https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/sigmoid_focal_loss_cuda_kernel.cuh#L11

**Comment by HawkRong** [comment]
> https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/sigmoid_focal_loss_cuda_kernel.cuh#L11

Ok, it's now clear. Thanks!
